VideoLocker/src/main/java/org/edx/mobile/view/adapters/DiscussionPostsAdapter.java

```diff
@@ -57,60 +68,63 @@ public void render(BaseViewHolder tag, DiscussionThread discussionThread) {
                 }
             } else {
                 icon = FontAwesomeIcons.fa_comments;
-                iconColor = edx_grayscale_neutral_light;
+                iconColor = (discussionThread.isRead() ? edx_grayscale_neutral_dark : edx_brand_primary_base);
             }
             holder.discussionPostTypeIcon.setIcon(icon);
             holder.discussionPostTypeIcon.setIconColor(iconColor);
         }
 
         {
-            String threadTitle = discussionThread.getTitle();
-            holder.discussionPostTitle.setText(threadTitle);
+            final SpannableString threadTitle = new SpannableString(discussionThread.getTitle());
+            if (!discussionThread.isRead()) {
+                CalligraphyTypefaceSpan boldSpan = new CalligraphyTypefaceSpan(openSansSemiBoldFont);
+                threadTitle.setSpan(boldSpan, 0, threadTitle.length(), Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+                holder.discussionPostTitle.setText(threadTitle);
```

Comment:
Why are you using spans instead of setting (or clearing) the typeface on the text view directly, which would be more efficient?

In any case, if you must do it this way, you can use the `CalligraphyUtils.applyTypefaceSpan()` utility method, which takes care of creating the typeface span, and also uses a global cache to hold spans for each typeface:

```` java
final CharSequence threadTitle = discussionThread.getTitle();
holder.discussionPostTitle.setText(discussionThread.isRead() ? threadTitle :
        CalligraphyUtils.applyTypefaceSpan(threadTitle, openSansSemiBoldFont));
````